master/pkg/model/task_container_defaults.go

```diff
@@ -60,8 +65,28 @@ func validatePortRange(portRange string) []error {
 	return errs
 }
 
+// UnmarshalJSON implements the json.Unmarshaler interface.
+// Setting defaults here is necessary over our usual "Define a default struct and unmarshal into it"
+// strategy because there are places (resource pool configs) where we need to know if the task
+// container defaults were set at all or if they were not; if they were set then that resource
+// pool's task container defaults are used instead of the toplevel master config's settings.  To
+// know if the user set them at the resource pool level, the resource pool has to have a nullable
+// pointer, which is not compatible with our usual strategy for defaults.
+func (c *TaskContainerDefaultsConfig) UnmarshalJSON(data []byte) error {
+	c.ShmSizeBytes = 4294967296
+	c.NetworkMode = "bridge"
+	type DefaultParser *TaskContainerDefaultsConfig
+	if err := json.Unmarshal(data, DefaultParser(c)); err != nil {
+		return errors.Wrap(err, "failed to parse task container defaults")
+	}
+	return nil
+}
+
 // Validate implements the check.Validatable interface.
-func (c TaskContainerDefaultsConfig) Validate() []error {
+func (c *TaskContainerDefaultsConfig) Validate() []error {
+	if c == nil {
```

Comment:
this is kinda funky, i imagine you agree. it's ok if it's nil in some places but in others it means it hasn't been defaulted right? but i guess it those places it couldn't be nil in this call, it would be a `&TaskContainerDefaultsConfig{}`, so it works, kinda feels like it barely correctly handles it in a complex way though. but it's ok for now.

Reply:
Indeed I do agree.